packages/supabase_flutter/lib/src/supabase_auth.dart

```diff
@@ -58,24 +58,26 @@ class SupabaseAuth with WidgetsBindingObserver {
       final persistedSession = await _localStorage.accessToken();
       if (persistedSession != null) {
         try {
-          Supabase.instance.client.auth.setInitialSession(persistedSession);
+          await Supabase.instance.client.auth
+              .setInitialSession(persistedSession);
+          shouldEmitInitialSession = false;
         } catch (error, stackTrace) {
           Supabase.instance.log(error.toString(), stackTrace);
         }
       }
     }
+    if (shouldEmitInitialSession) {
+      Supabase.instance.client.auth
+          // ignore: invalid_use_of_internal_member
+          .notifyAllSubscribers(AuthChangeEvent.initialSession);
+    }
```

Comment:
Moving this before the deep link handling to ensure the empty initial session is emitted first.